test/utils/TimeSlots.test.js

```diff
@@ -22,3 +22,81 @@ describe('getSlotMetrics', () => {
     expect(diff).toBe(60)
   })
 })
+
+describe('getRange', () => {
+  const min = dates.startOf(new Date(), 'day')
+  const max = dates.endOf(new Date(), 'day')
+  const slotMetrics = getSlotMetrics({ min, max, step: 60, timeslots: 1 })
+
+  test('getRange: 15 minute start of day appointment stays within calendar', () => {
+    let range = slotMetrics.getRange(
+      new Date(2018, 0, 29, 0, 0, 0),
+      new Date(2018, 0, 29, 0, 15, 0)
+    )
+    expect(range.top + range.height).toBeLessThan(100)
+    expect(range.height).toBeGreaterThan(0)
+  })
+
+  test('getRange: 1 hour start of day appointment stays within calendar', () => {
+    let range = slotMetrics.getRange(
+      new Date(2018, 0, 29, 0, 0, 0),
+      new Date(2018, 0, 29, 1, 0, 0)
+    )
+    expect(range.top + range.height).toBeLessThan(100)
+    expect(range.height).toBeGreaterThan(0)
+  })
+
+  test('getRange: 1 hour mid range appointment stays within calendar', () => {
+    let range = slotMetrics.getRange(
+      new Date(2018, 0, 29, 14, 0, 0),
+      new Date(2018, 0, 29, 15, 0, 0)
+    )
+    expect(range.top + range.height).toBeLessThan(100)
+    expect(range.height).toBeGreaterThan(0)
+  })
+
+  test('getRange: 3 hour mid range appointment stays within calendar', () => {
+    let range = slotMetrics.getRange(
+      new Date(2018, 0, 29, 14, 0, 0),
+      new Date(2018, 0, 29, 17, 0, 0)
+    )
+    expect(range.top + range.height).toBeLessThan(100)
+    expect(range.height).toBeGreaterThan(0)
+  })
+
+  test('getRange: full day appointment stays within calendar', () => {
+    let range = slotMetrics.getRange(
+      new Date(2018, 0, 29, 0, 0, 0),
+      new Date(2018, 0, 29, 23, 59, 0)
+    )
+    expect(range.top + range.height).toBeLessThan(100)
+    expect(range.height).toBeGreaterThan(0)
+  })
+
+  test('getRange: 1 hour end of day appointment stays within calendar', () => {
+    let range = slotMetrics.getRange(
+      new Date(2018, 0, 29, 23, 0, 0),
+      new Date(2018, 0, 29, 23, 59, 0)
+    )
+    expect(range.top + range.height).toBeLessThan(100)
+    expect(range.height).toBeGreaterThan(0)
+  })
+
+  test('getRange: 15 minute end of day appointment stays within calendar', () => {
+    let range = slotMetrics.getRange(
+      new Date(2018, 0, 29, 23, 45, 0),
+      new Date(2018, 0, 29, 23, 59, 0)
+    )
+    expect(range.top + range.height).toBeLessThan(100)
+    expect(range.height).toBeGreaterThan(0)
+  })
+
+  test('getRange: multi day appointment stays within calendar', () => {
+    let range = slotMetrics.getRange(
+      new Date(2018, 0, 29, 0, 0, 0),
+      new Date(2018, 0, 30, 4, 0, 0)
+    )
+    expect(range.top + range.height).toBeLessThan(100)
```

Comment:
Should this be equal to 100?

Reply:
I believe so, as there is code in the getRange that selects the max value between the start of the day and the start of the appointment. Along with the start of the day and the end of the appointment. My understanding is that this is what splits a multi day appointment over separate days in the week view when the `showMultiDayTimes` prop is set. 

I might be wrong though as this is just my understanding of how the calendar displays appointments